component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java

```diff
@@ -119,10 +119,10 @@ private void setUserAgentProperties(String userAgentHeader) {
     }
 
     private void copyDefaultPropertiesToRootLevel(Map<String, String> properties) {
-        String apiContext = properties.remove(Constants.API_CONTEXT);
-        properties.remove(Constants.USER_NAME);
-        eventMap.put(Constants.API_CONTEXT, apiContext);
-        eventMap.put(Constants.PROPERTIES, properties);
+
+        if (properties.get(Constants.API_CONTEXT) != null) {
+            eventMap.put(Constants.API_CONTEXT, properties.get(Constants.API_CONTEXT));
+        }
```

Comment:
this changes might break moseif integration. Hence shall we revert this?

Reply:
@pubudu538 can we remove this?
it was there from the beginning [1]

[1] https://github.com/wso2/apim-analytics-publisher/blob/1279c5ff78d251f90845375be90632ad9c487bc4/component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java#L124

Reply:
Fixed with https://github.com/wso2/apim-analytics-publisher/pull/96/commits/b58f780af5fb3e36c42831ab025087c5111e21d1